core/common/src/main/scala/net/liftweb/common/Box.scala

```diff
@@ -785,34 +822,124 @@ sealed abstract class Box[+A] extends Product with Serializable{
    * Although this function is different for true collections, because `Box` is
    * really a collection of 1, the two functions are identical.
    */
-  final def collectFirst[B](pf: PartialFunction[A, B]): Box[B] = {
+  def collectFirst[B](pf: PartialFunction[A, B]): Box[B] = {
     collect(pf)
   }
 }
 
 /**
+ * A `Full` or `Empty` box (stands in trivially for `Option`).
+ */
+sealed trait PresenceBox[+A] extends Box[A] {
+  override def isA[B](clsOrg: Class[B]): PresenceBox[B] = Empty
+  override def asA[B](implicit m: Manifest[B]): PresenceBox[B] = Empty
+
+  def or[B >: A](alternative: => Full[B]): Full[B]
+  def or[B >: A](alternative: => PresenceBox[B])(implicit a: DummyImplicit): PresenceBox[B]
+  def or[B >: A](alternative: => TryBox[B])(implicit a: DummyImplicit, b: DummyImplicit): TryBox[B]
+  def or[B >: A, E](alternative: => ParamTryBox[B, E])(implicit a: DummyImplicit, b: DummyImplicit, c: DummyImplicit): ParamTryBox[B, E]
+
+  override def map[B](f: A => B): PresenceBox[B] = Empty
+  def flatMap[B](f: A => PresenceBox[B]): PresenceBox[B]
+  override def collect[B](pf: PartialFunction[A, B]): PresenceBox[B] = ???
```

Comment:
Yeah, I'm not concerned about that. More about accident proofing the code. If there is a `???` implementation here, the compiler would say, for example, "Hey this interface is satisfied!" If someone were to somehow break the override of a child implementation, this would become the used implementation.

Whereas, if we _don't_ provide `???` here, breaking the override implementation should break the compile.